cdap-app-templates/cdap-etl/cdap-etl-realtime/src/test/java/co/cask/cdap/template/etl/realtime/RealtimeCubeSinkTest.java

```diff
@@ -108,22 +110,25 @@ public void test() throws Exception {
 
     AdapterManager manager = createAdapter(adapterId, adapterConfig);
 
-    long startTs = System.currentTimeMillis() / 1000;
+    final long startTs = System.currentTimeMillis() / 1000;
 
     manager.start();
-    // Let the worker run for 5 seconds
-    TimeUnit.SECONDS.sleep(5);
+    Tasks.waitFor(true, new Callable<Boolean>() {
+      @Override
+      public Boolean call() throws Exception {
+        DataSetManager<Cube> tableManager = getDataset("cube1");
```

Comment:
Can perhaps move this and the next line outside this method, so you don't have to redefine on line 128-129?